### PR TITLE
Added posts management (admin) create/edit/publish

### PR DESCRIPTION
**Pull request by ya-web**

### CODE REVIEW

File: src/Entity/Post.php [hunk 1]
```diff
@@ -3,23 +3,29 @@
...
 
 /**
+ * @Gedmo\SoftDeleteable(fieldName="deletedAt", timeAware=false, hardDelete=false)
  * @ORM\Entity(repositoryClass="App\Repository\PostRepository")
  */
 class Post
 {
+    public const STATUS_PENDING = 0;
+    public const STATUS_ACTIVE = 1;
+    public const STATUS_DELETED = 2;
```
Comment (kstefanini) [inline src/Entity/Post.php:17]: Seems redundant with `deletedAt`

File: src/Form/PostType.php [hunk 2]
```diff
@@ -16,22 +13,19 @@ class PostType extends AbstractType
...
-            ->add('title', TextType::class)
-            ->add('content', TextareaType::class)
-            ->add('submit', SubmitType::class)
         ;
     }
 
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults([
-            'data_class' => Post::class,
```
Comment (kstefanini) [inline src/Form/PostType.php:34]: Why do you remove this line?

File: src/Repository/PostRepository.php [hunk 3]
```diff
@@ -172,12 +175,16 @@ private function findNextOrPreviousPost(int $id, int $direction = 1): ?Post
     }
 
     /**
-     * @throws \Exception
```
Comment (kstefanini) [inline src/Repository/PostRepository.php:175]: For native Objects type you should keep the anti-slash and avoid `use`.

File: src/Repository/PostRepository.php [hunk 4]
```diff
@@ -172,12 +175,16 @@ private function findNextOrPreviousPost(int $id, int $direction = 1): ?Post
...
+     * @throws Exception
      */
     public function filterOutdatedEventscallback(Post $post): bool
     {
-        $validPost = ($post->getEndDate() >= (new \DateTime('now')));
+        if ($post->getEndDate()) {
+            $isValidPost = ($post->getEndDate() >= (new \DateTime('now')));
```
Comment (kstefanini) [inline src/Repository/PostRepository.php:183]: Maybe you should also check for endDate > startDate

**Review by kstefanini — APPROVED** [review]

File: src/Form/PostType.php (hunk 2 above)
Comment (ya-web) [reply src/Form/PostType.php:34]: this class is only used to reduce code duplication for forms as explained here https://symfony.com/doc/current/form/inherit_data_option.html

File: src/Entity/Post.php (hunk 1 above)
Comment (ya-web) [reply src/Entity/Post.php:17]: As we need status "pending", I thought we should complete the status for every situation

File: src/Repository/PostRepository.php (hunk 4 above)
Comment (ya-web) [reply src/Repository/PostRepository.php:183]: Isn't it pointless as EventPostType form builder constraints makes it impossible to record end date before start date?

Comment (kstefanini) [reply src/Repository/PostRepository.php:182]: Right, nevermind! :+1:  (Didn't understand this block)

File: src/Entity/Post.php (hunk 1 above)
Comment (kstefanini) [reply src/Entity/Post.php:17]: With `deletedAt` we already have status deleted info

Comment (ya-web) [reply src/Entity/Post.php:17]: ok but when I delete a post is the status (in database) supposed to stay "active" or "pending"? I think it should be "deleted"

**Review by kstefanini — APPROVED** [review]

**Pull request merged**